pykafka/cli/kafka_tools.py

```diff
@@ -54,11 +55,16 @@ def fetch_consumer_lag(client, topic, consumer_group):
     """
     latest_offsets = fetch_offsets(client, topic, 'latest')
     consumer = topic.get_simple_consumer(consumer_group=consumer_group,
-                                         auto_start=False, 
+                                         auto_start=False,
                                          reset_offset_on_fetch=False)
     current_offsets = consumer.fetch_offsets()
-    return {p_id: (latest_offsets[p_id].offset[0], res.offset)
-            for p_id, res in current_offsets}
+    consumer_id_dict = {}
+    for p_id, stat in current_offsets:
+        info = ast.literal_eval(stat[1])
+        consumer_id_dict[p_id] = [info['consumer_id'], info['hostname']]
```

Comment:
This line works only if the offset metadata stored in Kafka was put there by pykafka. Since this tools module is intended to be generically usable with any Kafka installation, the logic here needs to tolerate cases where the metadata was not written by pykafka. If we fail at either parsing the metadata as JSON or finding the expected `consumer_id` or `hostname` fields therein, we can simply pass with the assumption that the metadata was not written by pykafka.